subscriber_api: Convert module to TypeScript

LaPulgaaa opened the pull request:
<!-- Describe your pull request here.-->
subscriber_api: Convert module to TypeScript.


<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

andersk commented on web/src/subscriber_api.ts:
```diff
@@ -1,14 +1,19 @@
 import * as channel from "./channel";
 import * as people from "./people";
-
+import type {StreamSubscription} from "./sub_store";
```
The blank line you deleted was there to separate the imports from the comment; leave it there.

LaPulgaaa replied:
done @andersk . Apologies for some trivial mistakes. I have fixed the spacing after import and edited commit messages.
Thanks for your patience and guidance.

andersk reviewed (commented):
Do not close and reopen multiple pull requests for the same issue just to make changes. Instead, [update the existing pull request](https://zulip.readthedocs.io/en/latest/git/pull-requests.html#update-a-pull-request) in place (force-pushes are fine).

Please review our guidelines on [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html), then [tidy your commit history](https://zulip.readthedocs.io/en/latest/git/fixing-commits.html) and edit your commit message to conform to our [guidelines](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html#commit-messages).

LaPulgaaa commented:
@andersk  I have fixed the spacing, rebased my commit and edited the commit message according to the guidelines. Apologies for some trivial mistakes and thanks for the feedback .Looking forward to your review .!!

sayyedarib commented:
@LaPulgaaa I think it will be better to squash all the commits of Subscriber_api-ts into single commit.

LaPulgaaa commented:
@andersk , I think this file all is good now, I have squashed the commits into one to clean up things a bit.Let me know of any further improvements Sir.

andersk commented:
Your [current commit message](https://github.com/zulip/zulip/pull/28077/commits/24ef463484951425bfc00b3846f146b4762c22dd) is

> ts: Migrate subscriber_api.js to TypeScript
>
> subsriber_api-ts:Refractor Types
>
> subscriber_api-ts:Refract unnecessary types
>
> Subscriber_api-ts:Fixed spacing

which still doesn’t conform to our [guidelines](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html#commit-messages): “ts” is not a [part of the product](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html#commit-summary-part-1); the summary doesn’t [end with a period](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html#commit-summary-part-2); the description has three lines of [unnecessary personal narrative](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html#the-purpose-of-the-commit-description).

Our commit history is an important resource and it’s important to get these details right. Part of being an effective contributor is getting them right the first time, so the core maintainers don’t have to spend more time cleaning up your commits or helping you clean up your commits than it would have taken to just do the whole thing ourselves.

LaPulgaaa commented:
I am 100% with you on this @andersk . I guess I have to do a better job getting these details right!!!!!.

LaPulgaaa commented:
Thanks for the review @andersk .I have chopped off the vague commit discriptions and been a little more mindfull while writting the commit message and the PR discription. Will try to reduce reviewer's time spent per unit of my PRs.

LaPulgaaa commented:
Thanks for the review @andersk  .I have chopped off the vague commit discriptions and been a little more mindfull while writting the commit message and the PR discription. Will try to reduce amount of reviewer's time spent _per unit of my PR_.

andersk commented:
Merged, thanks @LaPulgaaa!
